### PR TITLE
add light and dark color scheme

### PR DESCRIPTION
**Pull request by sithlord48**
- Use System Color Scheme by default 
 - Add Option for Light and Dark theme 
 - Icons are detected based on theme 
   - *Where icons themes have support the fallback theme is set
   - My Window manager is SSD so it picks and draws the window titles and borders.

2022-08-20_Configuration_1-1

2022-08-20_Configuration_1-2

2022-08-20_Configuration_1-4

### CODE REVIEW

File: src/widgets/ConfigWindow.cpp [hunk 1]
```diff
@@ -417,7 +418,7 @@ void ConfigWindow::fillCharNameEdit()
 void ConfigWindow::setCharName(const QString &charName)
 {
 	int charId = listCharNames->currentIndex();
-	if (charId < 0 || charId > 8) {
+	if ( (charId < 0) || (charId > 8) ) {
```
Comment (myst6re) [inline src/widgets/ConfigWindow.cpp:421]: Please revert this change

File: src/widgets/ConfigWindow.cpp [hunk 2]
```diff
@@ -407,7 +408,7 @@ void ConfigWindow::resetColor()
 void ConfigWindow::fillCharNameEdit()
 {
 	int charId = listCharNames->currentIndex();
-	if (charId < 0 || charId > 8) {
+	if ( (charId < 0) || (charId > 8) ) {
```
Comment (myst6re) [inline src/widgets/ConfigWindow.cpp:411]: Please revert this change

File: src/core/SystemColor.cpp [hunk 3]
```diff
@@ -3,38 +3,34 @@
...
 #ifndef Q_OS_MAC
 
-#define systemColor(dark, light) \
-	bool darkTheme = Config::value("dark_theme", false).toBool(); \
-	return darkTheme ? dark : light;
-
 QColor SystemColor::red()
 {
-	systemColor(qRgb(0xF0, 0x90, 0x90), qRgb(0xb0, 0x18, 0x18))
+	return Config::inDarkMode() ? QColor(0xF0, 0x90, 0x90) : QColor(0xb0, 0x18, 0x18);
```
Comment (myst6re) [inline src/core/SystemColor.cpp:8]: Why not keep the macro?

File: src/widgets/ConfigWindow.cpp (hunk 1 above)
Comment (sithlord48) [reply src/widgets/ConfigWindow.cpp:421]: This resolves a warning is all

File: src/widgets/ConfigWindow.cpp (hunk 2 above)
Comment (sithlord48) [reply src/widgets/ConfigWindow.cpp:411]: same as above

File: src/core/SystemColor.cpp (hunk 3 above)
Comment (sithlord48) [reply src/core/SystemColor.cpp:8]: why use a macro when you can use a static method that works everywhere ?

Comment (myst6re) [reply src/core/SystemColor.cpp:8]: To prevent modify 7 lines like you do when you only need to change how dark and light is selected

A macro will not change the usage of the static method. You only avoid to repeat the same logic several times
